mmdet/models/detectors/lad.py

```diff
@@ -0,0 +1,87 @@
+import torch
+import torch.nn as nn
+from mmcv.runner import load_checkpoint
+
+from ..builder import DETECTORS, build_backbone, build_head, build_neck
+from .kd_one_stage import KnowledgeDistillationSingleStageDetector
+
+
+@DETECTORS.register_module()
+class LAD(KnowledgeDistillationSingleStageDetector):
+    """Implementation of `LAD <https://arxiv.org/pdf/2108.10520.pdf>`_."""
+
+    def __init__(self,
+                 backbone,
+                 neck,
+                 bbox_head,
+                 teacher_backbone,
+                 teacher_neck,
+                 teacher_bbox_head,
+                 teacher_ckpt,
+                 eval_teacher=True,
+                 train_cfg=None,
+                 test_cfg=None,
+                 pretrained=None):
+        super(KnowledgeDistillationSingleStageDetector,
+              self).__init__(backbone, neck, bbox_head, train_cfg, test_cfg,
+                             pretrained)
+        self.eval_teacher = eval_teacher
```

Comment:
No need, pass directly to the parent class.

Reply:
The parent class builds the teacher network via a config file:
```python
        self.teacher_model = build_detector(teacher_config['model'])
```
But my class builds the teacher network via components:
```python
        self.teacher_model = nn.Module()
        self.teacher_model.backbone = build_backbone(teacher_backbone)
        if teacher_neck is not None:
            self.teacher_model.neck = build_neck(teacher_neck)
        teacher_bbox_head.update(train_cfg=train_cfg)
        teacher_bbox_head.update(test_cfg=test_cfg)
        self.teacher_model.bbox_head = build_head(teacher_bbox_head)
```
That's why I re-write the `__init__` function. Do you have any better suggestions?